providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake_sql_api.py

```diff
@@ -232,29 +232,15 @@ def get_headers(self) -> dict[str, Any]:
         }
         return headers
 
-    def get_oauth_token(self, conn_config: dict[str, Any]) -> str:
+    def get_oauth_token(self, conn_config: dict[str, Any] | None = None) -> str:
         """Generate temporary OAuth access token using refresh token in connection details."""
-        url = f"{self.account_identifier}.snowflakecomputing.com/oauth/token-request"
-        data = {
-            "grant_type": "refresh_token",
-            "refresh_token": conn_config["refresh_token"],
-            "redirect_uri": conn_config.get("redirect_uri", "https://localhost.com"),
-        }
-        response = requests.post(
-            url,
-            data=data,
-            headers={
-                "Content-Type": "application/x-www-form-urlencoded",
-            },
-            auth=HTTPBasicAuth(conn_config["client_id"], conn_config["client_secret"]),  # type: ignore[arg-type]
+        warnings.warn(
+            "This method is deprecated and will be removed in Airflow 3.2. "
```

Comment:
`3.2` added thinking that should be enough to warn. Do we have a common approach for this?

Reply:
I don't follow why providers code raises warning of deprecation about Airflow core version

Reply:
My bad! Let me fix that! My idea was to add up 2 versions, by the way, if that is okay

Reply:
Fixed as `apache-airflow-providers-snowflake:6.4.0`